### ISSUE TITLE
No miss call notifications

### ISSUE THREAD

**Issue by vatsalchandan** [penn5/hwims-java#8]
No miss call notifications for missed calls ( don't know about rejected calls)

**Comment by penn5** [comment]
@vatsalchandan Missed calls when the phone is on sleep? Or when it is fully powered down?

**Comment by penn5** [comment]
@vatsalchandan found it. MmTelFeature#notifyRejectedCall(profile, reason) should do it.

**Comment by penn5** [comment]
I cannot do this without depending on https://android.googlesource.com/platform/frameworks/base/+/275b1358eba46f2eb7f27788ed4222134561889a%5E%21/telephony/java/android/telephony/ims/feature/MmTelFeature.java, which is not available until Q. I am unwilling to make this depend on Q so I won't add this feature yet, although I might link via reflection so that it can be optional

**Comment by penn5** [comment]
In the meantime I will make HwIms send a notification itself, rather than adding it to the call log

**Comment by vatsalchandan** [comment]
https://stackoverflow.com/questions/5910906/i-want-to-read-call-logs-in-the-android, https://developer.android.com/reference/android/provider/CallLog.Calls.html , this may help

**Comment by penn5** [comment]
Could do that but makes more sense to wait for Q to be standard. It has the APIs I need intergrated, thanks CAF

**Comment by Anutrix** [comment]
Any updates?

**Comment by penn5** [comment]
No. I broke my Huawei so if you want me to pick this up, buy me a new one.
This repo exists just for reference.
